### ISSUE TITLE
BUG: Check if Eventbrite issue is resolved

### ISSUE THREAD

**Issue by akaahanui** [DataKind-DC/capital-nature-ingest#175]
## Expected Behavior
We use the Eventbrite API for the following event sources. 

- Riverkeeper
- FONA
- Friends of Kenilworth
- NOVA Parks
- Sierra Club MD

## Current Behavior
Recently, Eventbrite made some changes that broke the `/search` endpoint. It was announced [here](https://groups.google.com/forum/#!topic/eventbrite-api/-E0MG7THMsc). But we use the `/organizers` endpoint, which nevertheless seems to be affected.

At present the scripts for each event source alternate between working and not working. It's very strange. For example, running `riverkeepr.py` will work one minute, returning the event found [here](https://www.eventbrite.com/o/anacostia-riverkeeper-10605256752) and then just a few seconds later, if you try it again, you get the following error:

```python
Traceback (most recent call last):
  File "sierra_club_md.py", line 185, in <module>
    events = main()
  File "sierra_club_md.py", line 147, in main
    fona_ingester.scrape()
  File "sierra_club_md.py", line 121, in scrape
    page = page.json()
  File "/c/Users/cmcallister/dev/capital-nature-ingest/env/lib/python3.6/site-packages/requests/models.py", line 897, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/lib/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.6/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

## Possible Solution
Read through every comment where Eventbrite announced the issues and see if there is some sort of rate limiting occurring. I did check this before but their API documentation was not up to date. Check again.

**Comment by csmcallister** [comment]
I'm working on #107 and that should shed some light. I'll circle back.

**Comment by csmcallister** [comment]
Just ran the riverkeeper event scraper by itself and it appears that the Eventbrite API is back up and running 🎉 

However, only one event was returned. I'll try to see if this checks out online.

**Comment by csmcallister** [comment]
And it does! So we should have this event source's events in the next run. 

Still need to see what's going on with the others though.

**Comment by csmcallister** [comment]
Eventbrite recently sent out the following email:

Hello,
We’re writing to inform you that access to the Eventbrite Event Search API (GET /v3/events/search/) will be shut down at 11:59 pm PT on Thursday, December 12, 2019.
We strongly encourage you to find and remove any code that makes requests to this Event Search API from your applications in advance.
Why is this happening?

We’re removing the Event Search API to further improve the Eventbrite platform and allow us to support more Creators and their events. Allowing public access to this particular API was impacting our platform and the high level of service we strive to provide to our creators and their attendees. We are able to provide alternative access to retrieve your event data through our Event APIs (see below).
 

What is the replacement API?

To get Events via our API, please see:

Retrieve an Event by ID - GET /v3/events/:event_id/
List Events by Venue - GET /v3/venues/:venue_id/events/
List Events by Organization - GET /v3/organizations/:organization_id/events/
If you’re retrieving private events on behalf of another user, you can complete the app authorization flow. If you’re interested in retrieving public events on behalf of many Eventbrite creators, you can apply to our distribution partner program.
We apologize for any inconvenience this change may cause, and thank you for your understanding. For questions, please reply to this email.


So it seems like we can just change the endpoint we use. Will try sometime this weekend.

**Comment by wkarney** [comment]
Reading the new API options above, it seems that there's no more **public** access to events unless you're a member of the [Eventbrite distribution partner program](https://www.eventbrite.com/l/distribution-partnerships/#form)? So it seems a little bit more than just changing the endpoint..am I reading that correctly?

**Comment by csmcallister** [comment]
I read that to say that a GET to `/v3/organizations/:organization_id/events/` was a part of the public API. Their [documentation](https://www.eventbrite.com/platform/api#/introduction/about-our-api) makes it seem that way.  The issue is that this new endpoint is unstable. I've tried on their [console](https://www.eventbrite.com/platform/api#/reference/event/update/list-events-by-organization?console=1)  with the following event organizers:
- [17820009959](https://www.eventbrite.com/o/potomac-riverkeeper-network-17820009959)
- [10605256752](https://www.eventbrite.com/o/anacostia-riverkeeper-10605256752)

You sometimes get a 404 saying that the organizer_id doesn't exist, despite the fact that it obviously does. And then you sometimes get a 200 without any events despite the fact that the event organizer pages clearly show upcoming events. 

My hope is that they're working on this issue and that the endpoint will eventually work. But given how they've handled this so far, I'm hesitant to spend time implementing this new endpoint only to see it never work. If, however, somebody else wanted to work on it -- hint, hint -- then I wouldn't stop them.

**Comment by csmcallister** [comment]
Just confirming that these issues still persist. The `17820009959` organization ID gets recognized by their API as a valid organization id, but the API doesn't return any events despite the fact that there are a bunch of organizations on their page. The other organization id (`10605256752`) isn't even recognized as a valid id by their API.

**Comment by csmcallister** [comment]
It doesn't look like Eventbrite is going to fix this, so we'll need to scrape the individual organizer pages on eventbrite.

**Comment by csmcallister** [comment]
@Frankie-Figz would you want me to assign this one to you?

**Comment by csmcallister** [comment]
Finally fixed @akaahanui ! Closed by d64cb2d
